pkg/runtime/runtime.go

```diff
@@ -881,21 +881,19 @@ func (a *DaprRuntime) isPubSubOperationAllowed(topic string, topicsList []string
 		if !inAllowedTopics {
 			return false
 		}
-	} else if len(topicsList) == 0 {
+	}
+	if len(scopedTopics) == 0 {
```

Comment:
take this condition in the beginning of the method and  return early..

Reply:
It should on check if its in allowedTopics first, there's a rule there that returns false in a certain condition.
Its ok to return true if that first check has passed.

Reply:
> It should first check if its in allowedTopics first, there's a rule there that returns false in a certain condition.
> Its ok to return true if that first check has passed.
It seems then these 2 independent code blocks are tied , could we possibly add the '&&' condition with inAllowedTopics == true while checking len(scopedTopics) == 0 that way it would be safe..